fw/connection.c

```diff
@@ -214,3 +214,22 @@ tfw_connection_hooks_unregister(int type)
 {
 	conn_hooks[TFW_CONN_TYPE2IDX(type)] = NULL;
 }
+
+/*
+ * Do an opposite to what tfw_connection_link_to_sk() does. Tempesta
+ * is unlinked from Sync Sockets layer, so that no data can be sent
+ * anymore on a connection. The previously held socket is released.
+ * Note that clearing of conn->sk is necessary. In case of failover
+ * on a server connection an indicator is needed to remove a hold
+ * on the socket. A zeroed conn->sk is that indicator.
+ */
+void
+tfw_connection_unlink_to_sk(TfwConn *conn)
```

Comment:
I understand the reason for the move, yet I wonder if there's a way to leave this function where it was, together with the opposite function. If there was it would be preferable.

Reply:
I think about it, but we need to tfw_classify_conn_close, which defined in http_limits.h. http_limits.h include connection.h, so I don't see any way to stay it in header except forward tfw_classify_conn_close declaration. I prefer move this function to connection.c